concent_api/core/views.py

```diff
@@ -1241,6 +1209,604 @@ def store_message_and_message_status(
+    assert isinstance(subtask_id,           str)
+    assert isinstance(provider_public_key,  bytes)
+    assert isinstance(requestor_public_key, bytes)
+    assert state in Subtask.SubtaskState
+
+    if state in Subtask.SUBTASK_ACTIVE_STATES:
+        assert next_deadline is not None
+    elif state in Subtask.SUBTASK_PASSIVE_STATES:
+        assert next_deadline is None
+
+    provider  = Client.objects.get_or_create_full_clean(provider_public_key)
+    requestor = Client.objects.get_or_create_full_clean(requestor_public_key)
+
+    subtask = Subtask(
+        task_id         = task_id,
+        subtask_id      = subtask_id,
+        provider        = provider,
+        requestor       = requestor,
+        state           = state.name,
+        next_deadline   = datetime.datetime.fromtimestamp(next_deadline, timezone.utc),
+    )
+
+    add_messages_to_subtask(
+        subtask,
+        task_to_compute,
+        report_computed_task,
+        force_get_task_result,
+        ack_report_computed_task,
+        reject_report_computed_task,
+        subtask_results_accepted,
+        subtask_results_rejected,
+    )
+
+    subtask.full_clean()
+    subtask.save()
+    return subtask
+
+
+def update_file_status(
+    client_public_key: bytes,
+):
+    force_get_task_result_list = PendingResponse.objects.filter(
+        Q(subtask__requestor__public_key = b64encode(client_public_key)) | Q(subtask__provider__public_key = b64encode(client_public_key)),
+        subtask__state                   = Subtask.SubtaskState.FORCING_RESULT_TRANSFER.name,  # pylint: disable=no-member
+    )
+
+    for pending_reponse in force_get_task_result_list:
+        report_computed_task    = deserialize_message(pending_reponse.subtask.report_computed_task.data.tobytes())
+        file_transfer_token     = create_file_transfer_token(
+            report_computed_task,
+            client_public_key,
+        )
+        if get_file_status(file_transfer_token):
+            subtask         = pending_reponse.subtask
+            subtask.state   = Subtask.SubtaskState.RESULT_UPLOADED.name  # pylint: disable=no-member
+            subtask.full_clean()
+            subtask.save()
+            mark_message_as_delivered(pending_reponse)
+
+            store_message_inbox(
+                response_type       = PendingResponse.ResponseType.ForceGetTaskResultUpload,
+                client_public_key   = subtask.provider.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.Receive,
+                subtask             = subtask,
+            )
+            store_message_inbox(
+                response_type       = PendingResponse.ResponseType.ForceGetTaskResultUpload,
+                client_public_key   = subtask.requestor.public_key_bytes,
+                queue               = PendingResponse.PendingResponseType.Receive,
+                subtask             = subtask,
+            )
+
+
+def update_concent_states(
```

Comment:
`update_concent_states`, `update_states_of_subtasks`, `update_subtask_state`, lets think about something more meaningful and more distinguishing.